torch/csrc/distributed/c10d/NCCLUtils.hpp

```diff
@@ -25,7 +25,8 @@ const inline char* getNcclErrorDetailStr(ncclResult_t error, c10::optional<std::
     case ncclUnhandledCudaError:
       return "ncclUnhandledCudaError: Call to CUDA function failed.";
     case ncclSystemError:
-      return "ncclSystemError: System call (socket, malloc, munmap, etc) failed.";
+      return "ncclSystemError: System call (e.g. socket, malloc) or external library call (e.g. ibverbs) failed or device error. "
+        "It can be also caused by unexpected exit of a remote peer, check NCCL warnings for peer id.";
```

Comment:
Should we mention this instead:

```
It can be also caused by unexpected exit of a remote peer, please check NCCL logs (after enabling NCCL_DEBUG=WARN or INFO) for the exact reason of the failure.
```